cairo/src/utils/signature.cairo

```diff
@@ -1,44 +1,364 @@
+from starkware.cairo.common.cairo_builtins import (
+    BitwiseBuiltin,
+    KeccakBuiltin,
+    ModBuiltin,
+    UInt384,
+    PoseidonBuiltin,
 )
+from starkware.cairo.common.poseidon_state import PoseidonBuiltinState
+from starkware.cairo.common.registers import get_fp_and_pc, get_label_location
+from ethereum.utils.numeric import divmod
+
 from starkware.cairo.common.math_cmp import RC_BOUND
-from starkware.cairo.common.cairo_secp.bigint import bigint_to_uint256, uint256_to_bigint
 from starkware.cairo.common.builtin_keccak.keccak import keccak_uint256s_bigend
-from starkware.cairo.common.cairo_secp.ec import ec_add, ec_mul, ec_negate
 from starkware.cairo.common.uint256 import Uint256
 from starkware.cairo.common.alloc import alloc
 from src.utils.maths import unsigned_div_rem
 
 from src.interfaces.interfaces import ICairo1Helpers
+from src.utils.circuit_basic_field_ops import div_mod_p, neg_mod_p, is_opposite_mod_p, is_eq_mod_p
+from src.utils.circuit_utils import (
+    N_LIMBS,
+    hash_full_transcript_and_get_Z_3_LIMBS,
+    scalar_to_epns,
+    felt_to_UInt384,
+    run_modulo_circuit_basic,
+)
+
+from src.utils.ecdsa_circuit import (
+    get_full_ecip_2P_circuit,
+    get_ADD_EC_POINT_circuit,
+    get_DOUBLE_EC_POINT_circuit,
+)
+
+struct G1Point {
+    x: UInt384,
+    y: UInt384,
+}
+
+namespace secp256k1 {
+    const CURVE_ID = 2;
+    const P0 = 0xfffffffffffffffefffffc2f;
+    const P1 = 0xffffffffffffffffffffffff;
+    const P2 = 0xffffffffffffffff;
+    const P3 = 0x0;
+    const N0 = 0xaf48a03bbfd25e8cd0364141;
+    const N1 = 0xfffffffffffffffebaaedce6;
+    const N2 = 0xffffffffffffffff;
+    const N_LOW_128 = 0xbaaedce6af48a03bbfd25e8cd0364141;
+    const N_HIGH_128 = 0xfffffffffffffffffffffffffffffffe;
+    const N3 = 0x0;
+    const A0 = 0x0;
+    const A1 = 0x0;
+    const A2 = 0x0;
+    const A3 = 0x0;
+    const B0 = 0x7;
+    const B1 = 0x0;
+    const B2 = 0x0;
+    const B3 = 0x0;
+    const G0 = 0x3;
+    const G1 = 0x0;
+    const G2 = 0x0;
+    const G3 = 0x0;
+    const MIN_ONE_D0 = 0xfffffffffffffffefffffc2e;
+    const MIN_ONE_D1 = 0xffffffffffffffffffffffff;
+    const MIN_ONE_D2 = 0xffffffffffffffff;
+    const MIN_ONE_D3 = 0x0;
+}
+
+const POW_2_32 = 2 ** 32;
+const POW_2_64 = 2 ** 64;
+const POW_2_96 = 2 ** 96;
+
+@known_ap_change
+func get_generator_point() -> (point: G1Point) {
+    // generator_point = (
+    //     0x79be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798,
+    //     0x483ada7726a3c4655da4fbfc0e1108a8fd17b448a68554199c47d08ffb10d4b8
+    // ).
+    return (
+        point=G1Point(
+            x=UInt384(
+                0x2dce28d959f2815b16f81798, 0x55a06295ce870b07029bfcdb, 0x79be667ef9dcbbac, 0x0
+            ),
+            y=UInt384(
+                0xa68554199c47d08ffb10d4b8, 0x5da4fbfc0e1108a8fd17b448, 0x483ada7726a3c465, 0x0
+            ),
+        ),
+    );
+}
+
+@known_ap_change
+func sign_to_UInt384_mod_secp256k1(sign: felt) -> (res: UInt384) {
```

Comment:
codebase specific, but return just the type, no need to make it a tuple (apply to all functions)

Reply:
`sign_to_uint384_mod_secp256k1`